tests/unit/lib/SGSparseMatrix_unittest.cc

```diff
@@ -23,22 +23,22 @@ using namespace shogun;
 #include <shogun/mathematics/eigen3.h>
 using namespace Eigen;
 
-//function to create sparse matrix
-template<typename T>
-SGSparseMatrix<T> GenerateSparseMatrix(const int32_t numberOfFeatures,
-                                       const int32_t numberOfVectors,
-                                       const float64_t sparseLevel)
+//function can generate both sparse and dense matrix accroding to sparse convention
+template<class MatrixType>
+void GenerateMatrixAccrodingToSparseConvention(const float64_t &sparseLevel,
+                                               const int32_t   &numberOfFeatures,
+                                               const int32_t   &numberOfVectors,
+                                               const int32_t   &randSeed,
+                                               MatrixType* matrix)
```

Comment:
@iglesias  yes, I know what you mean, and I also think that is what we should do. 

But the problem is if we initialize matrix as "MatrixType matrix(m,n)" and later use matrix(i,j) to access its contents. For sparse matrix, i goes from 0 to n-1, and j is from 0 to m-1, while dense matrix is i: 0 to m-1, j: 0 to n-1. 

This is how sparse and dense matrix is currently implemented and what I mean the sparse and dense matrix use different interfaces.